controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -92,20 +125,426 @@ func (r *StorageClassClaimReconciler) Reconcile(ctx context.Context, request rec
+		r.cephBlockPool.Namespace = r.namespace
+		r.cephBlockPool.Labels[controllers.StorageConsumerNameLabel] = r.storageConsumer.Name
+
+	} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
+		r.cephFilesystemSubVolumeGroup = &rookCephv1.CephFilesystemSubVolumeGroup{}
+		r.cephFilesystemSubVolumeGroup.Name = fmt.Sprintf("cephfilesystemsubvolumegroup-%s", r.storageConsumer.Name)
+		r.cephFilesystemSubVolumeGroup.Namespace = r.namespace
+	}
+
+	r.cephClientProvisioner = &rookCephv1.CephClient{}
+	r.cephClientProvisioner.Name = controllers.GenerateHashForCephClient(r.storageClassClaim.Name, "provisioner")
+	r.cephClientProvisioner.Namespace = r.namespace
+
+	r.cephClientNode = &rookCephv1.CephClient{}
+	r.cephClientNode.Name = controllers.GenerateHashForCephClient(r.storageClassClaim.Name, "node")
+	r.cephClientNode.Namespace = r.namespace
+
+	for _, cephResourceSpec := range r.storageClassClaim.Status.CephResources {
+		r.cephResourcesByName[cephResourceSpec.Name] = cephResourceSpec
+	}
+
+	r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateCreating)
+
+	if r.storageClassClaim.GetDeletionTimestamp().IsZero() {
+		if !controllers.Contains(r.storageClassClaim.GetFinalizers(), storageClassClaimFinalizer) {
+			r.log.Info("Finalizer not found for StorageClassClaim. Adding finalizer.", "StorageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+			r.storageClassClaim.SetFinalizers(append(r.storageClassClaim.GetFinalizers(), storageClassClaimFinalizer))
+			if err := r.update(r.storageClassClaim); err != nil {
+				r.log.Error(err, "Failed to update StorageClassClaim with finalizer.", "StorageClassClaim", klog.KRef(r.storageClassClaim.Namespace, r.storageClassClaim.Name))
+				return reconcile.Result{}, err
+			}
+		}
+		if r.storageClassClaim.Spec.Type == "blockpool" {
+
+			if err := r.reconcileCephClientRBDProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientRBDNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephBlockPool(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+		} else if r.storageClassClaim.Spec.Type == "sharedfilesystem" {
+			if err := r.reconcileCephClientCephFSProvisioner(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephClientCephFSNode(); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			if err := r.reconcileCephFilesystemSubVolumeGroup(); err != nil {
+				return reconcile.Result{}, err
+			}
+		}
+		cephResourcesReady := true
+		for _, cephResource := range r.storageClassClaim.Status.CephResources {
+			if cephResource.Phase != "Ready" {
+				cephResourcesReady = false
+				break
+			}
+		}
+
+		if cephResourcesReady {
+			r.storageClassClaim.Status.Phase = string(v1alpha1.StorageClassClaimStateReady)
+		}
+
+	} else {
+		// todo deletion of the Ceph Resources
+	}
```

Comment:
remove this else block. Add it when implementing delete.

Reply:
ack